src/main/java/org/betonquest/betonquest/api/quest/QuestTypeAPI.java

```diff
@@ -0,0 +1,105 @@
+package org.betonquest.betonquest.api.quest;
+
+import org.betonquest.betonquest.api.Objective;
+import org.betonquest.betonquest.api.profile.Profile;
+import org.betonquest.betonquest.id.ConditionID;
+import org.betonquest.betonquest.id.EventID;
+import org.betonquest.betonquest.id.ObjectiveID;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * The Core Quest Type logic.
+ */
+public interface QuestTypeAPI {
```

Comment:
Same here as with the `FeatureAPI` class. Either follow the InterfaceSegregationPrinciple and/or the Facade pattern. But don't define a Mega-Interface that will only ever be implemented once.